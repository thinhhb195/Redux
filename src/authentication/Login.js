import {connect} from "react-redux";
import {Navigate} from "react-router-dom";
import {useState} from "react";
import {handleLogin} from "../authentication/authedUser";
import 'bootstrap/dist/css/bootstrap.min.css';

const Login = ({dispatch, loggedIn}) => {
    const [username, setUsername] = useState("sarahedo");
    const [password, setPassword] = useState("sarahedo");

    if (loggedIn) {
        const urlParams = new URLSearchParams(window.location.search);
        const redirectUrl = urlParams.get('redirectTo');
        return <Navigate to={redirectUrl ? redirectUrl : "/"}/>;
    }

    const handleUsername = (e) => {
        const value = e.target.value;
        setUsername(value);
    };

    const handlePassword = (e) => {
        const value = e.target.value;
        setPassword(value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(handleLogin(username, password));
        setUsername("");
        setPassword("");
    };

    return (
        <div>
            <h1 className="text-3xl font-bold mt-9" data-testid="login-heading">Login</h1>
            <form onSubmit={handleSubmit}>
                <div>
                    <label htmlFor="username" className="block text-sm font-medium text-slate-700">Username</label>
                    <div className="mt-1">
                        <input
                            value={username}
                            onChange={handleUsername}
                            type="text"
                            name="username"
                            id="username"
                            data-testid="username"
                            className="mb3"/>
                    </div>
                </div>
                <div className="mt-6">
                    <label htmlFor="password" className="block text-sm font-medium text-slate-700">Password</label>
                    <div className="mt-1">
                        <input
                            value={password}
                            onChange={handlePassword}
                            type="password"
                            name="password"
                            id="password"
                            data-testid="password"
                            className="mb-3"/>
                    </div>
                </div>
                <div className="mt-6 text-right">
                    <button type="submit"
                            data-testid="submit"
                            className="w-100">
                        Login
                    </button>
                </div>
            </form>
        </div>
    );
};

const mapStateToProps = ({authedUser}) => ({
    loggedIn: !!authedUser,
});

export default connect(mapStateToProps)(Login);
