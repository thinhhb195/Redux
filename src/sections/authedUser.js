import {LOGOUT_AUTHED_USER, SET_AUTHED_USER} from "../authentication/authedUser";

export default function authedUser(state = null, action) {
  switch (action.type) {
    case SET_AUTHED_USER:
      return action.authedUser;
    case LOGOUT_AUTHED_USER:
      return null;
    default:
      return state;
  }
}
