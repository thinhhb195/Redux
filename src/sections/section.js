import {configureStore} from '@reduxjs/toolkit';
import authedUser from "./authedUser";
import users from "./users";
import questions from "./questions";

export const store = configureStore({
  reducer: {
    authedUser,
    users,
    questions,
  },
});
