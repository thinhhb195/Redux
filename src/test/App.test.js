import React from 'react';
import {render} from '@testing-library/react';
import App from '../App';
import {Provider} from "react-redux";
import {store} from "../sections/section";
import {BrowserRouter} from "react-router-dom";
import {setAuthedUser} from "../authentication/authedUser";

describe("App", () => {
    it("should render the component", () => {
        var component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );
                expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });
});
