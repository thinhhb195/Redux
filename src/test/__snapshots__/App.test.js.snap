// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto py-4"
      >
        <div>
          <h1
            class="text-3xl font-bold mt-9"
            data-testid="login-heading"
          >
            Login
          </h1>
          <form>
            <div>
              <label
                class="block text-sm font-medium text-slate-700"
                for="username"
              >
                Username
              </label>
              <div
                class="mt-1"
              >
                <input
                  class="mb3"
                  data-testid="username"
                  id="username"
                  name="username"
                  type="text"
                  value="sarahedo"
                />
              </div>
            </div>
            <div
              class="mt-6"
            >
              <label
                class="block text-sm font-medium text-slate-700"
                for="password"
              >
                Password
              </label>
              <div
                class="mt-1"
              >
                <input
                  class="mb-3"
                  data-testid="password"
                  id="password"
                  name="password"
                  type="password"
                  value="sarahedo"
                />
              </div>
            </div>
            <div
              class="mt-6 text-right"
            >
              <button
                class="w-100"
                data-testid="submit"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto py-4"
    >
      <div>
        <h1
          class="text-3xl font-bold mt-9"
          data-testid="login-heading"
        >
          Login
        </h1>
        <form>
          <div>
            <label
              class="block text-sm font-medium text-slate-700"
              for="username"
            >
              Username
            </label>
            <div
              class="mt-1"
            >
              <input
                class="mb3"
                data-testid="username"
                id="username"
                name="username"
                type="text"
                value="sarahedo"
              />
            </div>
          </div>
          <div
            class="mt-6"
          >
            <label
              class="block text-sm font-medium text-slate-700"
              for="password"
            >
              Password
            </label>
            <div
              class="mt-1"
            >
              <input
                class="mb-3"
                data-testid="password"
                id="password"
                name="password"
                type="password"
                value="sarahedo"
              />
            </div>
          </div>
          <div
            class="mt-6 text-right"
          >
            <button
              class="w-100"
              data-testid="submit"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
